src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,420 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang.reflect;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import static java.util.Map.entry;
+
+/**
+ * Represents a JVM access or module-related flag on a runtime member,
+ * such as a {@linkplain Class class}, {@linkplain Field field}, or
+ * {@linkplain Executable method}.
+ *
+ * <P>JVM access and module-related flags are related to, but distinct
+ * from Java language {@linkplain Modifier modifiers}. Some modifiers
+ * and access flags have a one-to-one correspondence, such as {@code
+ * public}. In other cases, some language-level modifiers do
+ * <em>not</em> have an access flag, such as {@code sealed} (JVMS
+ * {@jvms 4.7.31}) and some access flags have no corresponding
+ * modifier, such as {@linkplain #SYNTHETIC synthetic}.
+ *
+ * <p>The values for the constants representing the access and module
+ * flags are taken from sections of <cite>The Java Virtual Machine
+ * Specification</cite> including {@jvms 4.1} (class access and
+ * property modifiers), {@jvms 4.5} (field access and property flags),
+ * {@jvms 4.6} (method access and property flags), {@jvms 4.7.6}
+ * (nested class access and property flags), {@jvms 4.7.24} (method
+ * parameters), and {@jvms 4.7.25} (module flags and requires,
+ * exports, and opens flags).
+ *
+ * <p>The {@linkplain #mask() mask} values for the different access
+ * flags are <em>not</em> distinct. Flags are defined for different
+ * kinds of JVM structures and the same bit position has different
+ * meanings in different contexts. For example, {@code 0x0000_0040}
+ * indicates a {@link #VOLATILE volatile} field but a {@linkplain
+ * #BRIDGE bridge method}; {@code 0x0000_0080} indicates a {@link
+ * #TRANSIENT transient} field but a {@linkplain #VARARGS variable
+ * arity (varargs)} method.
+ *
+ * <p>The access flag constants are ordered by non-decreasing mask
```

Comment:
This paragraph seems more like an \@implNote, describing the ordering of the source.